src/main/java/com/zufar/onlinestore/security/jwt/JwtTokenValidator.java

```diff
@@ -19,13 +19,13 @@ public class JwtTokenValidator {
 
 	public boolean isValid(final String jwtToken) {
 		try {
-			final String username = jwtClaimExtractor.extractUsername(jwtToken);
+			final String email = jwtClaimExtractor.extractEmail(jwtToken);
 
-			final String usernameFromSystem = userDetailsService.loadUserByUsername(username).getUsername();
-			final String usernameFromJwtToken = jwtClaimExtractor.extractUsername(jwtToken);
+			final String emailFromSystem = userDetailsService.loadUserByUsername(email).getUsername();
```

Comment:
I can not understand this line.
You load full user info by calling "userDetailsService.loadUserByUsername(email)" and then you get "username" of this found user.
It always be the same email, because you are searching the user by some email (for example X) and if you find it in database, of course, you will get the same email X

Reply:
I think for self-signed JWT token it is enough to decode and check if user exist by decoded info (email in this case)